ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/handlers/help/BbeTopicsProvider.java

```diff
@@ -0,0 +1,97 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell.cli.handlers.help;
+
+import com.google.gson.Gson;
+import io.ballerina.shell.DiagnosticReporter;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Helper class to provide code completions related to /help topics.
+ *
+ */
+public class BbeTopicsProvider extends DiagnosticReporter {
+
+    private static final String BALLERINA_HOME = System.getProperty("ballerina.home");
+    private static final String EXAMPLES = "examples";
+    private static final String BBE_FILE = "index.json";
+    private static final String TOPICS = "topics";
+
+    private final List<String> topicList;;
+    private static BbeTopicsProvider bbeTopicsProvider = null;
+
+    private BbeTopicsProvider() {
+        topicList = new ArrayList<>();
+        populateTopicList();
+    }
+
+    private void populateTopicList() {
+        Gson gson = new Gson();
+
+        String file = BALLERINA_HOME + File.separator + EXAMPLES + File.separator + BBE_FILE;
+        String jsonString = readFileAsString(file);
+
+        if (jsonString != null) {
+            topicList.add(TOPICS);
+            jsonString = jsonString.trim();
+            BbeTitle[] bbeTitles = gson.fromJson(jsonString, BbeTitle[].class);
+            Stream<BbeTitle> streamList = Arrays.stream(bbeTitles);
+            streamList.forEach((bbeTitle) -> {
+                if (bbeTitle != null) {
+                    BbeRecord[] samples = bbeTitle.getSamples();
+                    Stream<BbeRecord> sampleList = Arrays.stream(samples);
+                    sampleList.forEach((bbeRecordElement) -> {
+                        if (bbeRecordElement != null) {
+                            topicList.add(bbeRecordElement.getUrl().replaceAll("-", " "));
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    public static synchronized BbeTopicsProvider getBbeTopicsProvider() {
+        if (bbeTopicsProvider == null) {
+                bbeTopicsProvider = new BbeTopicsProvider();
+        }
+        return bbeTopicsProvider;
+    }
+
+    public List<String> getTopicList() {
+        return topicList;
+    }
+
+    private static String readFileAsString(String file) {
+        String content;
+        try {
+            content = Files.readString(Paths.get(file));
+        } catch (IOException e) {
+            throw new RuntimeException("Error loading the file : " + e.getMessage(), e);
```

Comment:
How will this exception be handled? Why change to this from the debug diagnostic?

Reply:
Updated in 6d6f51ebb95023c10e576e195cc93ed3ad6e5ecf